lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -140,7 +140,7 @@ public function lastInsertId($name = null)
      */
     public function beginTransaction()
     {
-        db2_autocommit($this->conn, DB2_AUTOCOMMIT_OFF);
+        return db2_autocommit($this->conn, DB2_AUTOCOMMIT_OFF) === true;
```

Comment:
Why is `=== true` needed. Does it [return](https://www.php.net/db2_autocommit#refsect1-function.db2-autocommit-returnvalues) anything else than a boolean?

Reply:
As long as the second argument is given. But if it is omitted, the function will return an integer as far as I understand. Is this to be more explicit about the return value?

Reply:
Indeed. It's documented as `return @mixed` in the [stub](https://github.com/JetBrains/phpstorm-stubs/blob/f161ec38d8eb053d726d94c1c2ff961a848780fb/ibm_db2/ibm_db2.php).

I'd rather have an assertion instead of comparison. We shouldn't expect anything else than a boolean from a call with the second parameter.